agent/consul/server_test.go

```diff
@@ -2107,6 +2107,70 @@ func TestServer_hcpManager(t *testing.T) {
 
 }
 
+func TestServer_addServerTLSInfo(t *testing.T) {
+	testCases := map[string]struct {
+		expectErr   bool
+		tlsConfig   tlsutil.Config
+		checkStatus func(*testing.T, hcpclient.ServerStatus)
+	}{
+		"Success": {
+			tlsConfig: tlsutil.Config{
+				InternalRPC: tlsutil.ProtocolConfig{
+					CAFile:               "../../test/ca/root.cer",
+					CertFile:             "../../test/key/ourdomain.cer",
+					KeyFile:              "../../test/key/ourdomain.key",
+					VerifyIncoming:       true,
+					VerifyOutgoing:       true,
+					VerifyServerHostname: true,
+				},
+			},
+			checkStatus: func(t *testing.T, s hcpclient.ServerStatus) {
+				require.NotEmpty(t, s.ServerTLSMetadata.InternalRPC)
+				tlsInfo := s.ServerTLSMetadata.InternalRPC
+				require.True(t, tlsInfo.Enabled)
+				require.Equal(t, "test.internal", tlsInfo.CertIssuer)
+				require.Equal(t, "testco.internal", tlsInfo.CertName)
+				require.Equal(t, "40", tlsInfo.CertSerial)
+				require.NotEmpty(t, tlsInfo.CertExpiry)
+				require.True(t, tlsInfo.VerifyIncoming)
+				require.True(t, tlsInfo.VerifyOutgoing)
+				require.True(t, tlsInfo.VerifyServerHostname)
+
+				require.Len(t, tlsInfo.CertificateAuthorities, 1)
+				manualCAPem := tlsInfo.CertificateAuthorities[0]
+				require.NotEmpty(t, manualCAPem.CertExpiry)
+				require.Equal(t, manualCAPem.CertName, "test.internal")
+				require.NotEmpty(t, manualCAPem.CertSerial)
```

Comment:
nit: What if we defined an expected final struct and did a single `require.Equal` against that?

Reply:
Yea, good idea. I did the individual field checks to try to avoid hard-code the date fields, but everything else is hardcoded anyways. I think using an expected struct will be more readable 👍